lms/static/js/components/header/models/header.js

```diff
@@ -0,0 +1,17 @@
+/**
+ * A generic header model.
+ */
+;(function (define) {
+'use strict';
+define(['backbone'], function (Backbone) {
+    var HeaderModel = Backbone.Model.extend({
```

Comment:
Yeah, the intention is to be reusable, at least across the LMS. That's why the non-teams-specific parts are in `lms/static/js`.